Dynatrace/dynatrace-operator
# Remove custom Security Context Constraints

# Description
Fixes #1582

We had these SCCs for quite a while, but they mostly caused us/the-users problems. 
This is because we can't predict every possible OCP setup a user might have, so the following can (and did) happen:
- Our SCC no longer matched our component in the cluster, because the component was dynamically changed (e.g.: istio sidecar injection)
- Our SCC matched a user's component, restricting and causing it to fail

So to avoid further confusion and problems we are moving to use the built-in SCCs.
Our components have strict `securityContext` so they don't need to be further restricted, and its up to the customer to create an SCC if they wish to.

_Sidenote:_ This was already the case on OLM, as there you are not allowed to provide custom SCCs to your operator. 

**The default SCCs we will/want-to use:**
- `nonroot-v2`: 
  - For `operator/webhook/activegate` on OCP 4.11 or later
- `privileged`:
  - For `csi-driver/oneagent` on all OCP versions
    -  Also for operator/webhook/activegate on OCP 4.10 or earlier as we set a seccomp which is only allowed for `privileged` SCC on those version. (for some reason 🤷 )
       - So every component can use the `privileged` SCC, but everything uses the most restrictive SCC that is allowed to use. (so on newer OCP versions the more restricted SCC will be used)
       - This has the extra benefit that in case our components get dynamically modified we should still be fine, and the user can create an SCC accordingly. (previously the component wouldn't even show up, so there was no way to debug, what made the SCC no longer match)

## How can this be tested?
Deploy on OCP 4.11 or later, the operator/webhook/activegate should use `nonroot-v2`, the rest is `privileged`
Deploy on OCP 4.10 or earlier, everything should use `privileged`

## Checklist
